spring/src/main/java/org/axonframework/spring/eventsourcing/SpringAggregateSnapshotterFactoryBean.java

```diff
@@ -84,8 +90,18 @@ public SpringAggregateSnapshotter getObject() {
         TransactionManager txManager = transactionManager == null ? NoTransactionManager.INSTANCE :
                 new SpringTransactionManager(transactionManager, transactionDefinition);
 
-        SpringAggregateSnapshotter snapshotter = new SpringAggregateSnapshotter(eventStore, parameterResolverFactory,
-                                                                                executor, txManager, repositoryProvider);
+        if (deadlineManager == null) {
+            DeadlineTargetLoader deadlineTargetLoader = new DefaultDeadlineTargetLoader(configuration::repository,
```

Comment:
Nit: At this point you're insured that the `repositoryProvider` is non null, right?
Can't you just pass the `repositoryProvider` i.o. calling `configuration::repository`?